test/util/geo.test.cpp

```diff
@@ -6,6 +6,11 @@
 
 using namespace mbgl;
 
+TEST(LatLng, MixedCoordinateTypes) {
+    static_assert(std::is_assignable<LatLng, ScreenCoordinate>::value == false, "LatLng and ScreenCoordinate cannot be interchangeable.");
+    static_assert(std::is_assignable<LatLng, ProjectedMeters>::value == false, "LatLng and ProjectedMeters cannot be interchangeable.");
+}
```

Comment:
@jfirebaugh true - there is no simple way of forbidding implicit conversions like this besides deleting the base class assignment operator, which is not desired.